contribs/src/main/java/com/netflix/conductor/contribs/kafka/KafkaPublishTask.java

```diff
@@ -0,0 +1,271 @@
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.LongSerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+@Singleton
+public class KafkaPublishTask extends WorkflowSystemTask {
+
+	public static final String REQUEST_PARAMETER_NAME = "kafka_request";
+	public static final String NAME = "KAFKA_PUBLISH";
+	static final String MISSING_REQUEST = "Missing Kafka request. Task input MUST have a '" + REQUEST_PARAMETER_NAME + "' key with KafkaTask.Input as value. See documentation for KafkaTask for required input parameters";
+	private static final Logger logger = LoggerFactory.getLogger(KafkaPublishTask.class);
+	public static final String MISSING_BOOT_STRAP_SERVERS = "No boot strap servers specified";
+	public static final String MISSING_KAFKA_TOPIC = "Missing Kafka topic. See documentation for KafkaTask for required input parameters";
+	public static final String MISSING_KAFKA_BODY = "Missing Kafka body.  See documentation for KafkaTask for required input parameters";
+	public static final String FAILED_TO_INVOKE = "Failed to invoke kafka task due to: ";
+	protected ObjectMapper om = objectMapper();
+	protected Configuration config;
+	private String requestParameter;
+	KafkaProducerManager producerManager;
+
+
+	@Inject
+	public KafkaPublishTask(Configuration config, KafkaProducerManager clientManager) {
+		super(NAME);
+		this.config = config;
+		this.requestParameter = REQUEST_PARAMETER_NAME;
+		this.producerManager = clientManager;
+		logger.info("KafkaTask initialized...");
+	}
+
+	private static ObjectMapper objectMapper() {
+		final ObjectMapper om = new ObjectMapper();
+		om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		om.configure(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES, false);
+		om.configure(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES, false);
+		om.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+		om.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+		return om;
+	}
+
+	@Override
+	public void start(Workflow workflow, Task task, WorkflowExecutor executor) {
+
+		long taskStartMillis = Instant.now().toEpochMilli();
+		task.setWorkerId(config.getServerId());
+		Object request = task.getInputData().get(requestParameter);
+
+		if (Objects.isNull(request)) {
+			task.setReasonForIncompletion(MISSING_REQUEST);
```

Comment:
We can make one function to mark kafka task Failed and pass reason to avoid code duplicity. something like,
return markKafkaTaskFailed(String reason)